### PR TITLE
Doggo JSON cleanup

### LINKED ISSUE #62019: Small dogs still zombify, despite ``delete`` field used in their JSON definition
### Describe the bug

In #61668 I made the smallest dogs not zombify. Or, well, I thought I did. The code is there, it should work. It was such a simple change I didn't even bother to test it. The problem is that now it turns out my way to remove the ``zombifies_into`` field didn't work. This was brought to light in #62008 where it turned out that ``copy_from`` and ``zombifies_into`` have a deep romance that cannot be broken by conventional means. Using ``delete`` will simply be ignored, while supposedly (I did not test that myself, however, that is what I was told is the case) setting ``zombifies_into`` to ``mon_null`` _does_ stop the zombification while not removing the warning icon on a corpse.

### Steps to reproduce

1. Find a small dog like a beagle, a chihuahua or a dachshund
2. Murder
3. Notice the revival warning icon
4. Check the code and notice that the ``zombifies_into`` should be properly removed
5. Cry

### Expected behavior

``delete`` should not randomly lose its expected functionality on just one random field. I'd hazard a guess it doing just that isn't expected considering #61668 was approved and merged.

### Screenshots

A beautiful corpse of a dachshund with a revival warning icon despite the code implying the monster shouldn't zombify.
obraz

### Versions and configuration

- OS: Windows
    - OS Version: 10.0.19043.2130 (21H1)
- Game Version: aa2a276 [64-bit]
- Graphics Version: Tiles
- Game Language: English [en]
- Mods loaded: [
    Dark Days Ahead [dda],
    Disable NPC Needs [no_npc_food],
    No Fungal Growth [no_fungal_growth],
    Bionic Professions [package_bionic_professions],
    Tamable Wildlife [Tamable_Wildlife]
]

### Additional context

Didn't test it but I think it'd be worthwhile to take a look at what happens whether a monster using ``copy_from`` from some other monster that zombifies has a ``zombifies_into`` defined but aimed at a different zombie. Which zombie will it revive as?

### PR DESCRIPTION
**Pull request by VictoriaActual**
#### Summary
None

#### Purpose of change

Definitions for each breed of dog contained many duplicated entries and none of the puppies were using copy-from despite having lots of overlapping information. This was causing issues with a feature I was toying with as each dog was overwriting their flags instead of using "extend" or "delete".

#### Describe the solution

Add copy-from fields to each puppy type and removed duplicate entries in each dog breed.

#### Describe alternatives you've considered

Modifying each dog breed with the desired flags, but no thank you. That sounds like pain.

#### Testing
Spawned several dogs and tested that they could still be tamed and butchered into the right harvests. I tried it with labrador mutt, german shepard, plus the beagle and everything seemed to work smoothly with no changes.
After the changes to the "zombifies-into" field I tested killing a few of the puppy breeds and confirmed that they do not revive into zombie dogs as was intended. As per Karol's suggestion, this fixes #62019

### CODE REVIEW

File: data/json/monsters/mammal.json [hunk 1]
```diff
@@ -1571,83 +1295,31 @@
     "vision_night": 6,
     "harvest": "dog_tiny_with_skull",
     "fear_triggers": [ "HURT" ],
-    "delete": { "zombify_into": "mon_zombie_dog" },
+    "delete": { "zombify_into": "mon_zombie_dog", "flags": [ "SWARMS", "GROUP_MORALE" ] },
```
Comment (Karol1223) [inline data/json/monsters/mammal.json:1298]:
```suggestion
    "zombify_into": " ",
    "delete": { "flags": [ "SWARMS", "GROUP_MORALE" ] },
```

Can you do me a solid and switch the broken syntax for deleting zombification to this while you're here? ``delete`` doesn't work on non-arrays, so they still zombify. You gotta overwrite it with an empty field (not 100% sure I used an empty field in this example, this process is weird)

File: data/json/monsters/mammal.json [hunk 2]
```diff
@@ -1102,6 +1102,7 @@
     "vision_night": 4,
     "harvest": "dog_small_with_skull",
     "upgrades": { "age_grow": 42, "into": "mon_dog" },
+    "fear_triggers": [ "FRIEND_ATTACKED", "FRIEND_DIED", "HURT" ],
```
Comment (Karol1223) [inline data/json/monsters/mammal.json:1105]: This is cool but they still copy the anger triggers from their parent, so they now have the same triggers for fear and anger. You're gonna have to delete the anger triggers from them as well.

File: data/json/monsters/mammal.json [hunk 3]
```diff
@@ -1118,82 +1119,27 @@
...
-      "HIT_AND_RUN",
-      "KEENNOSE",
-      "PATH_AVOID_DANGER",
-      "SEES",
-      "SMELLS",
-      "WARM",
-      "CORNERED_FIGHTER",
-      "EATS"
-    ]
+    "delete": { "flags": [ "SWARMS", "GROUP_MORALE" ] }
```
Comment (Karol1223) [inline data/json/monsters/mammal.json:1126]: I wouldn't bother defining this. ``GROUP_MORALE`` fits dogs very well, at least.

**Review by Karol1223 — CHANGES_REQUESTED** [review]

**Comment by Karol1223** [comment]
Feel free to include 'Fixes #62019' in your PR description for a free serotonin boost

File: data/json/monsters/mammal.json (hunk 3 above)
Comment (BarndoBoi) [reply data/json/monsters/mammal.json]: I agree, I thought it was fairly off that only one or two breeds of dogs retained their pack stuff so I went ahead and kept the GROUP_MORALE and SWARMS flags on each breed and attempted to swap the fear and anger fields by overwriting the anger entry on the base pups. I'm about to test it now and see if I can piss off a whole swarm of doggos. I also changed the zombify-into section like you mentioned and I'm going to test that as well.
Something that is odd to me though is I haven't been able to spot what exactly the SWARMS flag does in any of the docs I've Ctrl+F'd my way through looking for it. :/

**Pull request merged**